### ISSUE TITLE
Too slow on YouTube videos 

### ISSUE THREAD

**Issue by naveedhassan913** [yausername/youtubedl-android#11]
It takes more than 10s to get video information.It starts receiving data after almost 4~5s .Can we set some options to increase speed. 

_Sent from my HOT 4 PRO using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by yausername** [comment]
Are you using the latest version? In that case I am not sure I can do much about that.

**Comment by naveedhassan913** [comment]
Yes latest

**Comment by xibr** [comment]
for me it takes 3 - 5 seconds.

**Comment by rahulsain-ingeniumedu** [comment]
any solution, I stream but yt videos load too slow at the speed of 50-150KB max

**Comment by xibr** [comment]
@rahulsain-ingeniumedu Related to  https://github.com/ytdl-org/youtube-dl/issues/29326

**Comment by chubecode** [comment]
I think we need some way to get just a part of information.
Like in my case, I just want to get the url, but it take ~ 10s to get all of data (incluled alot of things I don't need).
Hope it support way to input params define things we need and just return only it, I think it will be faster more than now.

**Comment by rahulsain** [comment]
> I think we need some way to get just a part of information. Like in my case, I just want to get the url, but it take ~ 10s to get all of data (incluled alot of things I don't need). Hope it support way to input params define things we need and just return only it, I think it will be faster more than now.

can we get absolute video URL right now?

**Comment by ConerK** [comment]
I also have the same problem, using the latest version, is there any way to solve it？

**Comment by xibr** [comment]
Is the problem still there?

**Comment by spiricn** [comment]
@xibr yes problem seems to be still there with the latest version(s). The same command on my PC finishes in ~1 second, while it takes close to 10 seconds when called via the library on an Android device.

Any ideas what might be responsible for the delay ?

**Comment by zaednasr** [comment]
@spiricn might be an android limitation. Even if you run yt-dlp in termux, its slow af there too. This library uses a termux library to run yt-dlp lol so it doesnt seem to run python in bare metal.
Not sure if its possible in android so that could be why its slow
